### PR TITLE
replicate Leaderboard page design, integrate sidebar, and ensure resp…

### PR DESCRIPTION
**Pull request by austineblaise**
Replicated the Leaderboard page design based on the provided Figma file.
Integrated the existing sidebar component from /frontend/app/app.
Ensured the page is fully mobile-responsive.
Screenshot from 2024-11-24 00-24-43
Screenshot from 2024-11-24 00-25-41
Screenshot from 2024-11-24 00-26-12
Screenshot from 2024-11-24 00-26-42

Screenshot from 2024-11-24 00-24-05

### CODE REVIEW

File: frontend/src/app/leaderboard/page.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  );
+}
+
+
+
+
+
+
+
+// "use client";
```
Comment (Dprof-in-tech) [inline frontend/src/app/leaderboard/page.tsx:39]: remove the commented code

File: frontend/src/app/leaderboard/dashboard.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        <div className="flex items-center justify-center gap-4">
+          <div className="flex flex-col items-start">
+            <p className="  text-black font-['Raleway'] ">Total participants</p>
+          </div>
+        </div>
+
+        <p className=" text-black font-['Raleway']">20,465,485 players</p>
+      </div>
+
+      <div className="w-full border-t-1 border-b-1 p-4 flex justify-between my-10 border rounded-xl  items-center bg-white">
```
Comment (Dprof-in-tech) [inline frontend/src/app/leaderboard/dashboard.tsx:27]: make this into a reuseable component that takes data from an array of objects

File: frontend/src/app/leaderboard/Nav.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
+"use client";
```
Comment (Dprof-in-tech) [inline frontend/src/app/leaderboard/Nav.tsx:1]: adapt this to the already exisiting nav used in /app/profile 

do not create a new one

File: frontend/public/images/badge.svg
Comment (Dprof-in-tech) [inline frontend/public/images/badge.svg:1]: this is already an asset as it is the peerprotocol logo, doesn't really matter as its just a place holder image

File: frontend/public/images/asset.png
Comment (Dprof-in-tech) [inline frontend/public/images/asset.png:1]: could you add your assets as svg instead of png? copy them as svg from the figma file

**Review by Dprof-in-tech — CHANGES_REQUESTED** [review]

File: frontend/src/app/leaderboard/sidebar.tsx
[the remark this answers is not in the document]
Comment (Dprof-in-tech) [reply frontend/src/app/leaderboard/sidebar.tsx:1]: also remove this sidebar on mobile view, it is not needed in mobile view

File: frontend/src/app/leaderboard/Nav.tsx (hunk 3 above)
Comment (austineblaise) [reply frontend/src/app/leaderboard/Nav.tsx:1]: I'm not entirely sure what is meant by "adapt this to the already existing nav used in /app/profile." The Figma design for the leaderboard navbar is different from the one in /app/profile.

File: frontend/src/app/leaderboard/sidebar.tsx
[the remark this answers is not in the document]
Comment (austineblaise) [reply frontend/src/app/leaderboard/sidebar.tsx:1]: the Figma design for the sidebar contains different content and structure compared to the existing sidebar (sidebar.tsx inside /app/app

**Comment by Dprof-in-tech** [comment]
a new change has been made to the quests folder structure, sync your fork with the main branch and you can find the new structure in /app/quests.

build your code into the relevant place. You have 6 hours to reply to this so i can confirm you are still working on it else your issue would be unassigned. Thank you

**Comment by austineblaise** [comment]
> a new change has been made to the quests folder structure, sync your fork with the main branch and you can find the new structure in /app/quests.
> 
> build your code into the relevant place. You have 6 hours to reply to this so i can confirm you are still working on it else your issue would be unassigned. Thank you

@Dprof-in-tech  Okay!!!!

**Comment by austineblaise** [comment]
@Dprof-in-tech I'm done !!!
Screenshot from 2024-11-25 12-58-10

File: frontend/package-lock.json
[the remark this answers is not in the document]
Comment (DanielEmmanuel1) [reply frontend/package-lock.json:1]: @austineblaise  You need to add this file `frontend/package-lock.json` to the `.gitignore`

**Review by Dprof-in-tech — APPROVED** [review]
Not as good as you kept making the same mistake across 3 commits, but i'll fix myself

**Pull request merged**